### ISSUE TITLE
SwaggerMappingError when not passing optional array-type query params

### ISSUE THREAD

**Issue by mxmul** [Yelp/bravado#202]
Using the latest version of bravado. I have a swagger endpoint with params that look something like this (snippet):

```json
 "parameters": [
     {
         "description": "The search term",
         "in": "query",
         "name": "term",
         "required": false,
         "type": "string"
     },
     {
         "default": [],
         "description": "The categories to filter on",
         "in": "query",
         "items": {
             "type": "string"
         },
         "name": "categories",
         "required": false,
         "type": "array"
     }
 ]
```

I'm able to curl the endpoint without specifying 'categories', but bravado has a fit if I don't explicitly pass an empty array:

```python
 >>> client = get_bravado_client()
 >>> client.search.search(term="pizza")
 Traceback (most recent call last):
   File "<stdin>", line 1, in <module>
   File "virtualenv_run/lib/python2.6/site-packages/bravado_decorators/header_decorator.py", line 22, in with_headers
     return getattr(self.resource, call_name)(*args, **kwargs)
   File "virtualenv_run/lib/python2.6/site-packages/bravado/client.py", line 250, in __call__
     self.operation, request_options, **op_kwargs)
   File "virtualenv_run/lib/python2.6/site-packages/bravado/client.py", line 291, in construct_request
     construct_params(operation, request, op_kwargs)
   File "virtualenv_run/lib/python2.6/site-packages/bravado/client.py", line 321, in construct_params
     marshal_param(remaining_param, None, request)
   File "virtualenv_run/lib/python2.6/site-packages/bravado_core/param.py", line 113, in marshal_param
     value = marshal_schema_object(swagger_spec, spec, value)
   File "virtualenv_run/lib/python2.6/site-packages/bravado_core/marshal.py", line 35, in marshal_schema_object
     return marshal_array(swagger_spec, schema_object_spec, value)
   File "virtualenv_run/lib/python2.6/site-packages/bravado_core/marshal.py", line 96, in marshal_array
     .format(type(array_value), array_value))
 bravado_core.exception.SwaggerMappingError: Expected list like type for <type 'NoneType'>: None
 >>> client.search.search(term='pizza', categories=[])
 <bravado.http_future.HttpFuture object at 0x2ba48d0>
```

**Comment by sjaensch** [comment]
Your default should be [] instead of "", since categories is an array and not a string.

**Comment by mxmul** [comment]
D'oh, you're right. The issue is still valid though – I get the same error after correcting my spec. Updated the issue.

**Comment by analogue** [comment]
Fixed in bravado-core 4.2.2
